OneCommunityGlobal/HighestGoodNetworkApp
# Sailavanya - Fix: Resolve blank page issue when selecting featured badges

# Description
This PR addresses a bug where the application would display a blank page when users attempted to save changes after selecting featured badges. The root cause was identified as a state update issue in the `BadgeReport.jsx` component. The `useEffect` hook was refactored to handle state updates correctly and to prevent memory leaks by ensuring cleanups.

## Related PRS (if any):
None

## Main changes explained:
- Updated the "BadgeReport.jsx" file to handle state updates properly.
- Added cleanup logic in the useEffect hook to avoid memory leaks.
- Ensured selected badges display correctly without triggering a blank page or crash.
- No changes were made to the functionality outside the scope of this bug fix.

## How to test:
1.Check out the `sailavanya_fix_blank_page_issue_fresh` branch.
2.Run "npm install" to ensure all dependencies are up to date.
3.Start the application locally using "npm start".
4.Log in as an admin user.
5.Navigate to the User Profile -> Featured Badges section.
6.Select or deselect badges and click on select featured button and click on save changes.
7.Verify that the badges are correctly saved and the page does not go blank.

## Screenshots or videos of changes:
Before the changes - https://www.loom.com/share/62f468163d844b1a8b07494804709d65
After the changes - https://www.loom.com/share/bd7d7ad9f532477789b368037f2281f4

## Note:
The export PDF functionality is outside the scope of this task and remains unchanged.
